proto/babylon/btcstkconsumer/v1/btcstkconsumer.proto

```diff
@@ -1,15 +1,49 @@
 syntax = "proto3";
 package babylon.btcstkconsumer.v1;
 
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
+
 option go_package = "github.com/babylonlabs-io/babylon/x/btcstkconsumer/types";
 
+// ConsumerType defines the type of consumer chain
+enum ConsumerType {
+  // COSMOS represents a Cosmos SDK chain consumer
+  COSMOS = 0;
+  // ETH_L2 represents an Ethereum L2 chain consumer
+  ETH_L2 = 1;
+}
+
 // ConsumerRegister is the registration information of a consumer
 message ConsumerRegister {
   // consumer_id is the ID of the consumer
   // - for Cosmos SDK chains, the consumer ID will be the IBC client ID
+  // - for ETH L2 chains, the consumer ID will be the chain ID of the ETH L2
+  //   chain
   string consumer_id = 1;
   // consumer_name is the name of the consumer
   string consumer_name = 2;
   // consumer_description is a description for the consumer (can be empty)
   string consumer_description = 3;
-}
\ No newline at end of file
+  // consumer_metadata is necessary metadata of the consumer, and the data
+  // depends on the type of integration
+  oneof consumer_metadata {
+    CosmosConsumerMetadata cosmos_consumer_metadata = 4;
+    ETHL2ConsumerMetadata eth_l2_consumer_metadata = 5;
+  };
+}
+
+// CosmosConsumerMetadata is the metadata for the Cosmos integration
+message CosmosConsumerMetadata {
+  // client_id defines the IBC client ID for the consumer chain
+  string client_id = 1;
```

Comment:
This is redundant with the main struct. Better to move `consumer_id` here, and replicate it in the Ethereum struct, with a proper description.

Moreover: Perhaps it's simpler / clearer to just make `ConsumerRegister` a `oneof`. At the expense of some redundancy in the consumer name and description fields.

Reply:
Yep this is redundant, let's remove this field. 

Personally I still prerfer to keep the current structure, as there might exist some applications that only care about id/name/description. We can follow up on this after backend/frontend comes up with concrete requirements.